handbook/engineering/distribution/product/roadmap.md

```diff
@@ -0,0 +1,253 @@
+1. (Q1 2020) [Automatic e2e testing](#automatic-e2e-testing)
+1. (TBD) [Automatic Docker image testing](#automatic-docker-image-testing)
+1. (TBD) [Upgrades across multiple Sourcegraph versions should be easier](#upgrades-across-multiple-sourcegraph-versions-should-be-easier)
+1. (TBD) [Sourcegraph should be released daily](#sourcegraph-should-be-released-daily)
+1. (TBD) [All site admins should have alerting set up to be notified when Sourcegraph is unhealthy](#all-site-admins-should-have-alerting-set-up-to-be-notified-when-sourcegraph-is-unhealthy)
+1. (TBD) [Push site admins to use Docker Compose or Kubernetes for production deployments](#push-site-admins-to-use-docker-compose-or-kubernetes-for-production-deployments)
+1. (TBD) [Add monitoring for common critical issues](#add-monitoring-for-common-critical-issues)
+1. (TBD) [Monitoring federation](#monitoring-federation)
+1. (TBD) [GitOps for all internal infrastructure](#gitops-for-all-internal-infrastructure)
+
+## Details (unordered)
+
+### Support customers in deploying Sourcegraph with 500k+ repositories
+
+We have had customers interested in deploying Sourcegraph at large-scale with ~500k+ repositories and will need to dedicate time to supporting them and making their trials go smoothly.
+
+- Owner: Uwe and Dave
+- Status: unplanned -> added unexpectedly to Q1 -> in-progress
+- [Tracking issue](https://github.com/sourcegraph/customer/issues/57)
+- Discussions: [Initial planning issue](https://github.com/sourcegraph/customer/issues/57), [discussion about costs at this scale](https://github.com/sourcegraph/customer/issues/20)
+- Dependencies: none
+
+### Kubernetes upgrades should have less merge conflicts
+
+Kubernetes upgrades involve a large number of merge conflicts today which are extremely time consuming and tedious for customers to resolve, preventing them from upgrading as frequently as they should be and creating a large and painful support burden for us.
+
+- Owner: Geoffrey
+- Status: planned for Q1 -> in-progress
+- [Tracking issue](https://github.com/orgs/sourcegraph/projects/68?card_filter_query=label%3Arfc-141)
+- Discussions: none
+- Dependencies: none
+
+### CI infrastructure that can run Docker containers in a reliable way
+
+Today we cannot release of Sourcegraph, run e2e tests, or perform Docker image tests in an automated fashion because our CI infrastructure does not support running Docker containers (or VMs/Vagrant) in a reliable way. Today we have a [side-car DIND container in our CI pipeline](https://sourcegraph.sgdev.org/search?q=repo:%5Esourcegraph/infrastructure%24+dind&patternType=literal) but it is flaky, unreliable, and a regular source of issues which has led to us removing automated testing (see [Automatic e2e testing](#automatic-e2e-testing)).
+
+- Owner: Stephen
+- Status: planning soon
+- [Tracking issue](https://github.com/sourcegraph/sourcegraph/issues/6887)
+- Discussions: none
+- Dependencies: none
+
+### Releasing Sourcegraph should be automated
```

Comment:
This should be #2 priority; should clarify that e2e and Docker image testing is dependency